src/main/java/com/revature/services/QuestionLogicImpl.java

```diff
@@ -32,12 +86,51 @@
 	@Override
 	public Page<QuestionPool> searchQuestions(Pageable pageable, String text, String subject) {
 		
-		if(subject.equalsIgnoreCase("all")){
+		if("all".equalsIgnoreCase(subject)){
 			return dao.findBySearchTerm(pageable, text);
 		} else {
 			return dao.findBySearchTerm(pageable, text, subject);
 		}
 
 	}
 	
+	@Override
+	public QuestionPool getQuestionById(Integer id) {
+		return dao.findOne(id);
+	}
+	
+	@Override
+	public QuestionComment getCommentById(Integer id) {
+		return commentDao.findOne(id);
+	}
+
+//UPDATE-----------------------------------
+	@Override
+	@Transactional
+	public QuestionPool updateQuestion(QuestionPool question) {
+		return dao.save(question);
+	}
+	
+	@Override
+	public QuestionComment updateComment(QuestionComment comment) {
+		return commentDao.save(comment);
+	}
+
+//DELETE-----------------------------------
+	@Override
+	@Transactional
+	public QuestionPool deleteQuestion(int id) {
+		QuestionPool question = dao.findOne(id);
+		dao.delete(question);
+		return question;
+	}
+
+	@Override
+	public QuestionComment deleteComment(int id) {
+		QuestionComment comment = commentDao.findOne(id);
+		commentDao.delete(comment);
+		System.out.println(comment);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS106)